nixos/modules/hardware/cpu/intel-npu.nix

```diff
@@ -0,0 +1,50 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.hardware.cpu.intel.npu;
+  type.driver =
+    with lib.types;
+    nullOr (enum [
+      "standalone"
+      # TODO: "with-compiler"
+    ]);
+in
+{
+  options = {
+    hardware.cpu.intel.npu = {
+      enable = lib.mkEnableOption "Intel NPU support";
+      driver = lib.mkOption {
+        type = type.driver;
+        default = "standalone";
+        description = ''
+          The driver to use for the Intel NPU.
+            - null: Only load the firmware.
+            - standalone: Use the standalone driver.
+
+          For more information, see:
+          https://github.com/intel/linux-npu-driver/blob/v1.13.0/docs/overview.md
+        '';
+      };
```

Comment:
IMO if people only want to load the firmware, they can just add ``hardware.firmware = [ pkgs.intel-npu-firmware ];`` to their config. Lets remove this option for now.